armory/datasets/package.py

```diff
@@ -18,16 +18,28 @@ def package(
     """
     Package a built dataset as .tar.gz, return path
     """
-    version, data_dir, built_data_dir, subdir = build.build_info(
+    version, data_dir, built_data_dir, subdir, builder_configs = build.build_info(
         name, version=version, data_dir=data_dir
     )
 
+    # print(builder_configs)
     data_dir = Path(data_dir)
-    expected_dir = data_dir / name / version
-    if not expected_dir.is_dir():
-        raise FileNotFoundError(f"Dataset {name} not found at {expected_dir}")
-    tar_full_filepath = common.get_cache_dataset_path(name, version)
 
+    if not builder_configs:
+        tar_list = [str(Path(name) / version)]
+    else:
+        # metadata.json contains default_config_name for the given dataset
+        tar_list = [
```

Comment:
the `+ [str(Path(name) / ".config")]` feels potentially dangerous, unless, do we know that this will _always_ exist for tfds datasets where `if builder_configs` is True? If not, the `FileNotFoundError` below will get raised

Could we programatically add the contents inside `Path(name)` to the tarlist, perhaps obviating the need for the if/else logic and the hardcoded addition of `.config`?

Reply:
`.config` and `metadata.json` is hardcoded in https://github.com/tensorflow/datasets/blob/v4.6.0/tensorflow_datasets/core/dataset_builder.py#L1257-L1286 and not saved as an accessible attribute - unless `tfds` developers change how `default_config_name` is implemented, don't think we have a choice

Reply:
have we tested what happens if we don't include this `.config/` dir? Does tfds error out trying to reach somewhere that doesn't exist? Including it is adding complexity to our code, so just making sure it's warranted

Reply:
I think it will work without `.config/metadata.json` - the only potential issue now is we won't necessarily know what the default dataset will be if `config` is not specified in the armory config file

Reply:
do you mind adding a pointer to the tfds link above in the Armory comment?

Reply:
resolved with f146e0b2